app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/PdfViewer.kt

```diff
@@ -1,217 +1,442 @@
+            tint = MaterialTheme.colorScheme.onSurface)
+      },
+      onClick = {
+        // Unselects the note and navigates back to the previous screen
+        navigationActions.goBack()
+        noteViewModel.selectedNote(null)
+      })
+}
+
+/**
+ * Composable function that creates a row item for the PDF viewer screen.
+ *
+ * @param buttonText The text to display in the row item.
+ * @param textColor The color of the text.
+ * @param testTag The test tag for the row item.
+ * @param onClick The lambda function to call when the row item is clicked.
+ * @param icon The icon to display in the row item.
+ */
+@Composable
+fun RowItemPdfViewer(
+    buttonText: String,
+    textColor: Color = MaterialTheme.colorScheme.onSurface,
+    testTag: String,
+    onClick: () -> Unit = {},
+    icon: @Composable () -> Unit
+) {
+  Row(
+      modifier = Modifier.fillMaxWidth().clickable { onClick() }.padding(10.dp).testTag(testTag),
+      content = {
+        icon()
+        Spacer(Modifier.padding(10.dp))
+        Text(text = buttonText, color = textColor, style = MaterialTheme.typography.titleMedium)
+      })
+}
+
+/**
+ * Function to update (append) the MD file with the extracted text from the PDF file. If the MD file
+ * does not exist, it creates a new one. After updating the MD file, it navigates to the Edit
+ * Markdown screen.
+ *
+ * @param text The extracted text from the PDF file.
+ * @param noteViewModel The NoteViewModel to access the selected note.
+ * @param fileViewModel The FileViewModel to update the MD file.
+ * @param navigationActions The NavigationActions object to navigate between screens.
+ * @param context The context used to access the cache directory.
+ */
+fun updateMdFile(
+    text: String,
+    noteViewModel: NoteViewModel,
+    fileViewModel: FileViewModel,
+    navigationActions: NavigationActions,
+    context: Context
+) {
+  val note = noteViewModel.selectedNote.value
+
+  fileViewModel.downloadFile(
+      uid = note!!.id,
+      fileType = FileType.NOTE_TEXT,
+      context = context,
+      onSuccess = { mdFile ->
+        mdFile.appendText("\n" + text)
+        fileViewModel.updateFile(
+            uid = note.id,
+            fileUri = Uri.fromFile(mdFile),
+            fileType = FileType.NOTE_TEXT,
+            onSuccess = { navigationActions.navigateToAndPop(Screen.EDIT_NOTE_MARKDOWN) })
+      },
+      onFileNotFound = {
+        val mdFile =
+            File(context.cacheDir, "${note.id}.md").apply { if (!exists()) createNewFile() }
+        mdFile.appendText(text)
+        fileViewModel.updateFile(
+            uid = note.id,
+            fileUri = Uri.fromFile(mdFile),
+            fileType = FileType.NOTE_TEXT,
+            onSuccess = { navigationActions.navigateToAndPop(Screen.EDIT_NOTE_MARKDOWN) })
+      },
+      onFailure = {
+        Log.e("PdfViewerScreen", "Error downloading MD file: $it")
+        Toast.makeText(context, "Error: markdown file could not be updated", Toast.LENGTH_LONG)

```

Comment:
Please avoid hard coded strings and use string resources instead.

Reply:
I think `Toast`s do not use string resources yet. But I can update it 👍